File: 21-templates/npm-package/templates.ts
import axios from 'axios';
const API = 'https://cost-katana-backend.store/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function createTemplate(name: string, template: string) {
  const res = await axios.post(`${API}/prompt-templates`, {name, template},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Template created:', res.data.data.id);
  return res.data.data;
}

File: 21-templates/python-sdk/templates.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def create_template(name, template):
    res = requests.post(f"{API}/prompt-templates",
        json={"name": name, "template": template},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"✅ Template created: {res.json()['data']['id']}")
    return res.json()["data"]
